.github/workflows/zxf-prepare-extended-test-suite.yaml

```
@@ -38,20 +36,38 @@ jobs:
   tag-for-xts:
     name: Tag for XTS promotion
     runs-on: network-node-linux-medium
-    if: ${{ github.event.workflow_run.conclusion == 'success' && !github.event.workflow_run.head_repository.fork && github.event.workflow_run.head_branch == 'develop'}}
+    if: ${{ !github.event.workflow_dispatch.repository.fork }}
     steps:
       - name: Harden Runner
         uses: step-security/harden-runner@f086349bfa2bd1361f7909c78558e816508cdc10 # v2.8.0
         with:
           egress-policy: audit
 
+      - name: Checkout Default Branch
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        with:
+          fetch-depth: '0'
+          ref: 'develop'
+          token: ${{ secrets.GH_ACCESS_TOKEN }}
+
+      - name: Validate Input Ref
+        id: validate-input
+        env:
+          COMMIT_ID: ${{ inputs.ref }}
+        run: |
+          if [[ 0 -eq $(git merge-base --is-ancestor $COMMIT_ID develop) ]]; then
```

The code appears as if it intends to check the process exit code; however, this is instead checking the output from `stdout` against the value of `0` and does not check the exit code. 

```suggestion
          if git merge-base --is-ancestor "${COMMIT_ID}" develop >/dev/null 2>&1; then
```